lib/base-compiler.js

```diff
@@ -146,7 +157,18 @@ Compile.prototype.prepareArguments = function (userOptions, filters, inputFilena
     return options.concat(userOptions || []).concat([this.filename(inputFilename)]);
 };
 
-Compile.prototype.compile = function (source, options, filters) {
+Compile.prototype.generateAST = function(inputFilename, options) {
+    // These options make Clang produce an AST dump
+    var newOptions = options.concat(["-Xclang", "-ast-dump", "-fsyntax-only"]);
+
+    execOptions = this.getDefaultExecOptions();
+    // A higher max output is needed for when the user includes headers
+    execOptions.maxOutput = 1024*1024*1024 ;
```

Comment:
A gigabyte is quite a high limit here: we might have to revisit this if we start OOMing CE :)